State decisions

wtbutler opened the pull request:
Added transitions to coast and brake, added logic to transition to coast and brake automatically.
Also added unit testing for new transitions. 

The cement has cracks
From the powerful whacks
Made by forehead smacks
Because my braincase lacks

wtbutler committed:
Step One of coast/break transitions

I think I added transition commands for coast and
break, need to test and double ckeck.

wtbutler committed:
Added test for trans_coast

Doesn't work right now, looking in to fixing it.

wtbutler committed:
Tests are fixed

I descovered that I forgot the name of the
command id not 5 minutes after making it.

wtbutler committed:
Finished tests

Test cases for Flight_Coast and Flight_Brake
have been added. All tests are successful.

wtbutler committed:
Added logic for flight control

wtbutler committed:
Fixed the issues in Pod_State

Actually added the transitions from Accel and Coast.
Removed unimplemented function from MotionModel.hpp

wtbutler committed:
Fixed tests

schmidtwmark commented on CentralComputing/Pod_State.cpp:
```diff
@@ -209,13 +213,54 @@ void Pod_State::steady_launch_ready(std::shared_ptr<NetworkManager::Network_Comm
+	double IDEAL_DECCEL = 9.8;
+	double LENGTH_OF_TRACK = 1000;
+	double BUFFER_LENGTH = 20;
+	double MAX_VELOCITY = 550;
```
Let's move these constants out of the function, put them at the top of this file for now

schmidtwmark commented on CentralComputing/Pod_State.cpp:
```diff
@@ -209,13 +213,54 @@ void Pod_State::steady_launch_ready(std::shared_ptr<NetworkManager::Network_Comm
+	double vel = state->x[1];
+	double acc = state->x[2];
+	
+	if (-.5*vel*vel/IDEAL_DECCEL+vel*vel/IDEAL_DECCEL >= LENGTH_OF_TRACK - BUFFER_LENGTH - pos) {
```
Refactor this into "ShouldBrake(double pos, double vel)"

schmidtwmark replied:
We should probably set up some unit tests to test this function for certain known values

schmidtwmark reviewed (changes requested):
Looks good, but please address these changes

wtbutler committed:
Cleaned up PodState a little

Still need to add unit tests, but refactored some functionality
into ShouldBrake and moved constants to a more general location.

wtbutler committed:
Added busy wait

Also added tests for Busy Wait in UtilsTest.cpp

wtbutler committed:
Revert "Cleaned up PodState a little"

This reverts commit e176687e1cce7df36e2f5b2359a1236f48998bd1.

wtbutler committed:
Revert "Revert "Cleaned up PodState a little""

This reverts commit 49628c267bd9302a2336c458ae4003e5798d6808.

wtbutler committed:
Revert "Added busy wait"

This reverts commit 499049488c04eae5472ba6436149626bdfad40c3.
